File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
'''
t 문자열을 리스트에 담기
p는 int로 변환
p의 길이를 for문으로 돌려 t[:i] 이런식으로 나눠보기
'''
def solution(t, p):
    answer = 0
   
    for i in range(len(t)):
        a = t[i:i+len(p)]
        
        if len(a) == len(p):
            if t[i:i+len(p)] <= p:
                
                answer += 1
            
    # print(a)
        
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int tLength = t.length();
        int pLength = p.length();

        
        for (int i = 0; i <= tLength - pLength; i++) {
            String substring = t.substring(i, i + pLength);
            if (substring.compareTo(p) <= 0) {
                answer++;
            }
        }

        return answer;
    }
}